### ISSUE TITLE
Add pycharm like check rule about "Remove redundant parentheses"

### ISSUE THREAD

**Issue by andyxning** [astral-sh/ruff#19861]
### Summary

Pycharm by default will warning about `Remove redundant parentheses` just like this:

```python

def a() -> tuple[int, int]:
    return (1, 1)
```

**Comment by MichaReiser** [comment]
I'll merge this into an existing issue that also requests for a rule that flags redundant parentheses (they use a different example but the request is the same)
